api/grpcserver/globalstate_service.go

```diff
@@ -274,13 +278,25 @@ func (s GlobalStateService) AccountDataStream(in *pb.AccountDataStreamRequest, s
 		channelReceipt chan events.TxReceipt
 	)
 	if filterAccount {
-		channelAccount = events.GetAccountChannel()
+		if s.accountChannel == nil {
+			channelAccount = events.SubscribeToAccounts(30)
+		} else {
+			channelAccount = s.accountChannel
+		}
 	}
 	if filterReward {
-		channelReward = events.GetRewardChannel()
+		if s.rewardsChannel == nil {
+			channelReward = events.SubscribeToRewards(30)
```

Comment:
It looks like you're not handling the error case here, where this returns nil, as you are elsewhere